platform/features/layout/src/LayoutController.js

```diff
@@ -127,6 +127,25 @@ define(
                 e.preventDefault();
             }
 
+            function getComposition(domainObject){
+                return domainObject.useCapability('composition');
+            }
+
+            function composeView (composition){
+                $scope.composition = composition;
+                return composition.map(function (object) {
+                        return object.getId();
+                    }) || [];
+            }
+
+            //Will fetch fully contextualized composed objects, and populate
+            // scope with them.
+            function refreshComposition() {
+                return getComposition($scope.domainObject)
+                    .then(composeView)
+                    .then(self.layoutPanels);
+            }
```

Comment:
> Basically I ignore the ids from the watch, because I subsequently fetch the composition from a capability (because it's no longer provided on $scope), and there is no guarantee the ids will match. So instead I just use the ids returned from the capability since that is the composition that is actually displayed in the layout.

Ah, that makes sense. While I _think_ it would be safe to assume IDs won't change (or that the watch will be invoked again when they do), agree that the approach you're using makes it easier to be confident of correctness.